### PR TITLE
Revise ruby valid logic to one line

### PR DESCRIPTION
**Pull request by dav1a1223**
Dear @FGRibreau , 
When I use this handy gem and read the source code,
I found the logic of `valid?` and `valid_email?` a little complex.
So I took a look at `.py` and revise the logic to one line.
Hoping that make it easier to understand :)
I'm a beginner to contribute open source, please give me your advice. :)
Thx a lot!!

### CODE REVIEW

File: platform/ruby/mail_checker.rb [hunk 1]
```diff
@@ -6,15 +6,11 @@ module MailChecker
...
 
   def self.valid?(email)
-    return false unless valid_email?(email)
-
-    !blacklisted?(email)
+    return valid_email?(email) && !blacklisted?(email)
```
Comment (JulienBreux) [inline platform/ruby/mail_checker.rb]:
```ruby
valid_email?(email) && !blacklisted?(email)
```

File: platform/ruby/mail_checker.rb [hunk 2]
```diff
@@ -6,15 +6,11 @@ module MailChecker
...
   end
 
   def self.valid_email?(email)
-    return false if email.nil?
-
-    email =~ EMAIL_REGEX
+    return !email.nil? && !!(email =~ EMAIL_REGEX)
```
Comment (JulienBreux) [inline platform/ruby/mail_checker.rb]:
```ruby
!email.nil? && !!(email =~ EMAIL_REGEX)
```

File: platform/ruby/mail_checker.tmpl.rb [hunk 3]
```diff
@@ -6,15 +6,11 @@ module MailChecker
   BLACKLIST = [{{& listSTR }}].to_set
 
   def self.valid?(email)
-    return false unless valid_email?(email)
-
-    !blacklisted?(email)
+    return valid_email?(email) && !blacklisted?(email)
```
Comment (JulienBreux) [inline platform/ruby/mail_checker.tmpl.rb]:
```ruby
valid_email?(email) && !blacklisted?(email)
```

File: platform/ruby/mail_checker.tmpl.rb [hunk 4]
```diff
@@ -6,15 +6,11 @@ module MailChecker
...
   end
 
   def self.valid_email?(email)
-    return false if email.nil?
-
-    email =~ EMAIL_REGEX
+    return !email.nil? && !!(email =~ EMAIL_REGEX)
```
Comment (JulienBreux) [inline platform/ruby/mail_checker.tmpl.rb]:
```ruby
!email.nil? && !!(email =~ EMAIL_REGEX)
```

**Comment by FGRibreau** [comment]
Hello @dav1a1223, 

as @JulienBreux said, just remove the return and I will accept this PR :)

**Comment by dav1a1223** [comment]
OK, it's done. @FGRibreau @JulienBreux thanks a lot !!

**Comment by FGRibreau** [comment]
@dav1a1223 nice work, released in v3.0.16 :)

**Pull request merged**